File: app/src/main/java/com/reednit/android/network/json/ListingDataJson.java
package com.reednit.android.network.json;

import com.reednit.android.room.Link;
import com.squareup.moshi.Json;

import java.util.ArrayList;
import java.util.List;

class ListingDataJson {

    @Json(name = "children")
    private List<LinkJson> children = null;

    public List<LinkJson> getChildren() { return children; }

//    ContentValues[] toContentValuesArray() {
//
//        ContentValues[] result = new ContentValues[children.size()];
//
//        int index = 0;
//        for(LinkJson child : children){
//            result[index] = child.toContentValues();
////            result[index].put(LocalContract.LinkEntry.COLUMN_NAME_ORDINAL, index);
//            index ++;
//        }
//
//        return result;
//
//    }

    List<Link> toModel(){
        List<Link> result = new ArrayList<>();
        for(LinkJson child : children){
            Link model = child.toModel();
            if(model != null)
                result.add(child.toModel());
        }
        return result;
    }

}

File: app/src/main/java/com/reednit/android/repository/remote/json/ListingDataJson.kt
package com.reednit.android.repository.remote.json

import com.squareup.moshi.Json

data class ListingDataJson(
        @Json(name = "children")
        var children: List<LinkJson>
)

//package com.reednit.android.repository.remote.json;
//
//import com.reednit.android.repository.local.Link;
//import com.squareup.moshi.Json;
//
//import java.util.ArrayList;
//import java.util.List;
//
//@SuppressWarnings("unused")
//class ListingDataJson {
//
//    @Json(name = "children")
//    private List<LinkJson> children = null;
//
//    public List<LinkJson> getChildren() { return children; }
//
//    List<Link> toModel(){
//        List<Link> result = new ArrayList<>();
//        for(LinkJson child : children){
//            Link model = child.toModel();
//            if(model != null)
//                result.add(child.toModel());
//        }
//        return result;
//    }
//
//}
